submitter/tx_submitter.go

```diff
@@ -34,119 +36,108 @@ func NewTxSubmitter(cfg *config.Config, executor *executor.Executor, daoManager
 }
 
 func (s *TxSubmitter) SubmitTransactionLoop() {
+	s.cachedEventHash = make(map[uint64][]byte, math.MaxInt)
+	submitAttempt := 0
```

Comment:
As we discussed, the logic will change a bit after the in-turn challenger on chain.